google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -268,12 +267,14 @@ public QueryResult getQueryResults(QueryResultsOption... options) {
       throw new UnsupportedOperationException(
           "Getting query results is supported only for " + Type.QUERY + " jobs");
     }
-
-    // TODO(pongad): check job stats.state instead.
-    QueryResponse response = bigquery.getQueryResults(getJobId(), options);
-    if (!response.jobCompleted()) {
-      throw new IllegalStateException("the job hasn't completed yet");
+    if (getStatus().getState() != JobStatus.State.DONE) {
+      throw new BigQueryException(BigQueryException.UNKNOWN_CODE, "the job hasn't completed yet");
     }
+    if (getStatus().getError() != null) {
```

Comment:
I assume `getError()` corresponds to [job.status.errorResult](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#status.errorResult)?

Reply:
Yes it is.